spec/API_specification/array_api/array_object.py

```diff
@@ -346,7 +361,7 @@ def __float__(self: array, /) -> float:
         Parameters
         ----------
         self: array
-            zero-dimensional array instance. Must have a real-valued floating-point data type.
+            zero-dimensional array instance. Should have a real-valued floating-point data type.
```

Comment:
Technically, conversion must be refused for complex, I think (Python and NumPy do this, and I suspect everyone else also).
"Should" indicates a suggestion, I would suggest watering it down to saying that it is implementation defined. (I would lean towards saying that it should be allowed for all numeric dtypes except that it must not be for complex).

Reply:
Heh I wanted to avoid deliberation but yes if even my half-measure here this needs clarification then I might need to go the full monty...